pkg/migration/controllers/clusterpair.go

```diff
@@ -218,10 +218,29 @@ func getClusterPairSchedulerStatus(clusterPairName string, namespace string) (st
 }
 
 func (c *ClusterPairController) cleanup(clusterPair *stork_api.ClusterPair) error {
+	skipDelete := false
+	cpName := ""
 	if clusterPair.Status.RemoteStorageID != "" {
+		// verify if any other cluster pair using the same RemoteStorageID
+		cpList, err := storkops.Instance().ListClusterPairs("")
+		if err != nil {
+			return err
+		}
+		for _, cp := range cpList.Items {
+			// No need to handle current reconciled clusterpair
+			// Since clusterPair will have deleteTimeStamp set and will be ignored
+			if cp.Status.RemoteStorageID == clusterPair.Status.RemoteStorageID &&
+				cp.DeletionTimestamp == nil {
+				skipDelete = true
+				cpName = cp.Namespace + "/" + cp.Name
+				break
+			}
+		}
+	}
+	if !skipDelete {
 		return c.volDriver.DeletePair(clusterPair)
 	}
-	return nil
+	return fmt.Errorf("underlying storage pair is being used by another clusterpair: %s", cpName)
```

Comment:
If you return an error here, the stork ClusterPair CR will not deleted. IMO we should let the CR get deleted. Usually someone would be deleting a namespace and the cluster pair not getting deleted will block it.